cli/internal/cmd/upgradeapply.go

```diff
@@ -244,16 +245,32 @@ func parseTerraformUpgradeVars(cmd *cobra.Command, conf *config.Config, fetcher
 		imageRef = strings.Replace(imageRef, "Versions", "versions", 1)
 
 		vars := &terraform.AzureClusterVariables{
-			CommonVariables:      commonVariables,
-			Location:             conf.Provider.Azure.Location,
+			Name:                 conf.Name,
 			ResourceGroup:        conf.Provider.Azure.ResourceGroup,
 			UserAssignedIdentity: conf.Provider.Azure.UserAssignedIdentity,
-			InstanceType:         conf.Provider.Azure.InstanceType,
-			StateDiskType:        conf.Provider.Azure.StateDiskType,
 			ImageID:              imageRef,
-			SecureBoot:           *conf.Provider.Azure.SecureBoot,
-			CreateMAA:            conf.GetAttestationConfig().GetVariant().Equal(variant.AzureSEVSNP{}),
-			Debug:                conf.IsDebugCluster(),
+			NodeGroups: map[string]terraform.AzureNodeGroup{
+				"control_plane_default": {
+					Role:          "ControlPlane",
+					InstanceCount: 1, // TODO
```

Comment:
Left-over `TODO` ?

Reply:
in the past we didn't set this during an upgrade, so it assumed the type default of 0 and it worked.. maybe 0 just assumes 1? I actually wonder why this was not a problem if we created a cluster with a count higher than 1